src/Marten/Schema/Identity/Sequences/SequenceFactory.cs

```diff
@@ -55,17 +55,20 @@ public void WritePermissions(DdlRules rules, StringWriter writer)
 
         public ISequence SequenceFor(Type documentType)
         {
-            return _sequences.GetOrAdd(documentType, type =>
-            {
-                var settings = _options.Storage.MappingFor(type).HiloSettings ?? _options.HiloSequenceDefaults;
-                return new HiloSequence(_tenant, _options, documentType.Name, settings);
-            });
+            return Hilo(documentType, _options.Storage.MappingFor(documentType).HiloSettings ?? _options.HiloSequenceDefaults);
         }
 
         public ISequence Hilo(Type documentType, HiloSettings settings)
         {
-            return _sequences.GetOrAdd(documentType,
-                type => new HiloSequence(_tenant, _options, documentType.Name, settings));
+            return _sequences.GetOrAdd(GetSequenceName(documentType, settings),
+                sequence => new HiloSequence(_tenant, _options, sequence, settings));
+        }
+
+        private string GetSequenceName(Type documentType, HiloSettings settings)
+        {
+            if (!string.IsNullOrEmpty(settings.SequenceName))
```

Comment:
**QUESTION:** Is it doublechecking if the `SequenceName` might be `string.Empty` or is it something that may happen? If it's doublechecking then maybe it would be good to verify also if it's not whitespace?

Reply:
@oskardudycz Default behavior is that the `SequenceName` is not set and will be `null` as per the initialization. So checking for `!string.IsNullOrEmpty(settings.SequenceName)` is good enough. I agree that we have to validate the sequence name for the below
> Sequence name is a string of 31 or fewer characters[1]. A name must start with a letter or an underscore; the rest of the string can contain letters, digits, and underscores.

Reply:
There's no validation for a default sequencename (equals to the <documentype>.name)

ps. we don't talk about postgresql sequences, but the value of the sequece_name column in the table hilo

Reply:
Understood, please ignore my comment in the validation part